mp/src/game/client/momentum/fx_mom_explosions.cpp

```diff
@@ -0,0 +1,175 @@
+//========= Copyright � 1996-2006, Valve Corporation, All rights reserved. ============//
+//
+// Purpose: Game-specific explosion effects
+//
+//=============================================================================//
+#include "cbase.h"
+#include "c_basetempentity.h"
+#include "c_te_effect_dispatch.h"
+#include "c_te_legacytempents.h"
+#include "dlight.h"
+#include "engine/IEngineSound.h"
+#include "iefx.h"
+#include "mom_shareddefs.h"
+#include "tempent.h"
+#include "tier0/vprof.h"
+#include "weapon/mom_weapon_parse.h"
+#include "weapon/weapon_mom_rocketlauncher.h"
+
+CWeaponInfo *GetMomWeaponInfo(int iWeapon)
+{
+    const char *pszWeaponAlias = WeaponIDToAlias(iWeapon);
+    if (!pszWeaponAlias)
+    {
+        return NULL;
+    }
+
+    WEAPON_FILE_INFO_HANDLE hWpnInfo = LookupWeaponInfoSlot(pszWeaponAlias);
+    if (hWpnInfo == GetInvalidWeaponInfoHandle())
+    {
+        return NULL;
+    }
+
+    CWeaponInfo *pWeaponInfo = static_cast<CWeaponInfo *>(GetFileWeaponInfoFromHandle(hWpnInfo));
+    return pWeaponInfo;
+}
+
+void TFExplosionCallback(const Vector &vecOrigin, const Vector &vecNormal, CWeaponID iWeaponID,
+                         ClientEntityHandle_t hEntity)
+{
+    CWeaponInfo *pWeaponInfo = NULL;
+    pWeaponInfo = GetWeaponInfo(iWeaponID);
```

Comment:
You can join declaration and assignments if you aren't doing anything fancy between them.
`const auto pWeaponInfo = GetWeaponInfo(iWeaponID);`